File: HolidayTestApp/app/src/main/java/com/yammobots/holidaytestapp/di/ViewModelFactoryModule.java
package com.yammobots.holidaytestapp.di;

import androidx.lifecycle.ViewModelProvider;

import com.yammobots.holidaytestapp.viewmodels.ViewModelProviderFactory;

import dagger.Binds;
import dagger.Module;

/**
 * Created by kaungkhantsoe on 2020-02-02.
 **/

@Module
public abstract class ViewModelFactoryModule {

    @Binds
    public abstract ViewModelProvider.Factory bindViewModelProviderFactory(ViewModelProviderFactory viewModelProviderFactory);

   /*

   // Works the same as " bindViewModelProviderFactory "
    @Provides
    static ViewModelProvider.Factory bindFactory(ViewModelProviderFactory viewModelProviderFactory) {
        return viewModelProviderFactory;
    }

    */



}

File: HolidayTestApp/app/src/main/java/com/yammobots/holidaytestapp/di/ViewModelFactoryModule.kt
package com.yammobots.holidaytestapp.di

import androidx.lifecycle.ViewModelProvider
import com.yammobots.holidaytestapp.viewmodels.ViewModelProviderFactory
import dagger.Binds
import dagger.Module

/**
 * Created by kaungkhantsoe on 22/06/2020.
 **/

@Module
abstract class ViewModelFactoryModule {
    @Binds
    abstract fun bindViewModelProviderFactory(viewModelProviderFactory: ViewModelProviderFactory?): ViewModelProvider.Factory? /*

   // Works the same as " bindViewModelProviderFactory "
    @Provides
    static ViewModelProvider.Factory bindFactory(ViewModelProviderFactory viewModelProviderFactory) {
        return viewModelProviderFactory;
    }

    */
}
